lib/string.js

```diff
@@ -406,9 +406,20 @@ internals.String = class extends Any {
         });
     }
 
-    base64() {
+    base64(padding) {
 
-        const regex = /^(?:[A-Za-z0-9+\/]{4})*(?:[A-Za-z0-9+\/]{2}==|[A-Za-z0-9+\/]{3}=)?$/;
+        if (padding) {
+            Hoek.assert(typeof padding === 'boolean', 'padding requirement must be boolean');
+        }
+        const paddingRequired = (typeof padding === 'undefined') ? true : padding;
```

Comment:
This won't catch when a `null` is passed. In general I would usually handle this with something like the following:

```js
padding = padding === false ? padding : padding || true;
```